pkg/api/2018-09-30-preview/default.go

```diff
@@ -17,6 +17,22 @@ func setDefaults(oc *OpenShiftManagedCluster) {
 		oc.Properties.MasterPoolProfile.Count = to.Int64Ptr(3)
 	}
 
+	if len(oc.Properties.AgentPoolProfiles) == 0 {
+		oc.Properties.AgentPoolProfiles = []AgentPoolProfile{
+			{
+				Name:  to.StringPtr("infra"),
+				Count: to.Int64Ptr(3),
```

Comment:
Staring at this, I wonder if we should default this count to 3 in all cases (and not just when no AgentPoolProfiles were specified)?

Reply:
Yes - the logic's not quite right.  It should ensure an infra APP exists, and set its count to 3 if it is nil.